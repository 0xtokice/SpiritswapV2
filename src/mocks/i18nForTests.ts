/* istanbul ignore next */
export const mockUseTranslation = () => ({
  t: key => key,
  i18n: {
    language: undefined,
  },
});
