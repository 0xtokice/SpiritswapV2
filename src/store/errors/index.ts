import { createSlice } from '@reduxjs/toolkit';
import { ErrorState } from './errors';

const initialState: ErrorState = {
  unexpectedError: false,
};

const errorReducers = createSlice({
  name: 'errorReducer',
  initialState,
  reducers: {
    setUnexpectedError: (state, action) => {
      state.unexpectedError = action.payload;
    },
  },
});

// Actions
export const { setUnexpectedError } = errorReducers.actions;
export default errorReducers.reducer;
