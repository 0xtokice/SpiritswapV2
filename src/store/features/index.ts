import { createSlice } from '@reduxjs/toolkit';
import { FeatureFlags } from './features';

const initialState: FeatureFlags = {
  environment: `${process.env.NODE_ENV}`,
  development: {
    apemode: false,
  },
  production: {
    apemode: false,
  },
};

const featureReducers = createSlice({
  name: 'featureReducer',
  initialState,
  reducers: {},
});

export default featureReducers.reducer;
