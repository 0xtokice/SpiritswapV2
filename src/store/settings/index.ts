import { createSlice } from '@reduxjs/toolkit';
import { Settings } from './settings';
import { isMobile } from 'utils/isMobile';

const initialState: Settings = {
  notifications: !isMobile(),
  suggestions: true,
  keepSuggestions: false,
};

const settingsReducer = createSlice({
  name: 'settingsReducer',
  initialState,
  reducers: {
    setUserNotifications: (state, { payload }) => {
      state.notifications = payload.notifications;
    },
    setUserSuggestions: (state, { payload }) => {
      state.suggestions = payload.suggestions;
    },
    setKeepSuggestions: (state, { payload }) => {
      state.keepSuggestions = payload.keepSuggestions;
    },
  },
});

export const { setUserNotifications, setUserSuggestions, setKeepSuggestions } =
  settingsReducer.actions;

export default settingsReducer.reducer;
