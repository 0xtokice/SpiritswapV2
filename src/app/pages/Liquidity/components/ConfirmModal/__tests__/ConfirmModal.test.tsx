import { render } from '@testing-library/react';
import { ChakraThemeWrapper } from 'app/components/shared/testing/ChakraTheme';
import { ThemeWrapper } from 'app/components/shared/testing/ThemeWrapper';
import { mockUseTranslation } from 'mocks/i18nForTests';
import { store } from 'store';
import { Provider } from 'react-redux';
import { ConfirmModal } from '..';

jest.mock('react-i18next', () => ({
  useTranslation: () => mockUseTranslation(),
}));

const onCancel = () => {};
const onConfirm = () => {};

const parsedAmounts = {
  SPIRIT: 1000,
  WBTC: 2000,
};

const Component = () => {
  return (
    <ChakraThemeWrapper>
      <ThemeWrapper>
        <Provider store={store}>
          <ConfirmModal
            onCancel={onCancel}
            onConfirm={onConfirm}
            sharePool={0.003}
            parsedAmounts={parsedAmounts}
            price={'4000'}
            setZapDirectly={() => {}}
            zapDirectly={false}
          />
        </Provider>
      </ThemeWrapper>
    </ChakraThemeWrapper>
  );
};

describe('Confirm Modal component', () => {
  test('renders without crashing', () => {
    const { container } = render(<Component />);
    expect(container.firstChild).toBeTruthy();
  });

  it('should render and match snapshot', () => {
    const { container } = render(<Component />);
    expect(container).toMatchSnapshot();
  });
});
