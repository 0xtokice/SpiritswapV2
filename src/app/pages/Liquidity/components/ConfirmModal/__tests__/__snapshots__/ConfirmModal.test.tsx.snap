// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirm Modal component should render and match snapshot 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  cursor: pointer;
}

.c0,
.c0 > div,
.c0 > svg,
.c0 > img {
  width: xl;
  height: xl;
}

<div>
  <div
    class="css-1kq59qq"
  >
    <div
      class="css-1ui6izz"
    >
      <button
        class="chakra-button css-ftvr69"
        type="button"
      >
        <div
          class="c0 css-2q6ptx"
        >
          <svg
            class="chakra-icon css-18w02b3"
            focusable="false"
          >
            back.svg
          </svg>
        </div>
      </button>
      <h2
        class="css-1io1c3r"
      >
        liquidity.confirmPanel.confirmAddingLiquidity
      </h2>
    </div>
    <div
      class="css-1ia939w"
    >
      <p
        class="chakra-text css-0"
      >
        liquidity.confirmPanel.youPay
      </p>
    </div>
    <div
      class="css-j7qwjs"
    >
      <p
        class="chakra-text css-xlrm81"
      >
        liquidity.common.youWillReceive
      </p>
      <div
        class="css-clmqzu"
      />
      <p
        class="chakra-text css-mlm88v"
      >
        4000
         
      </p>
      <div
        class="css-f808a9"
      >
        <svg
          class="chakra-icon css-1tcckqc"
          focusable="false"
        >
          danger.svg
        </svg>
        <p
          class="chakra-text css-1cpcgfy"
        >
          liquidity.confirmPanel.alert
        </p>
      </div>
    </div>
    <div
      class="css-1uzrj0v"
    >
      <div
        class="css-z7mtfw"
      >
        <p
          class="chakra-text css-nfvc7l"
        >
          liquidity.confirmPanel.shareOfPool
        </p>
        <p
          class="chakra-text css-1wuxeh2"
        >
          &lt;0.01
        </p>
      </div>
      <div
        class="chakra-stack css-p7rcp5"
      >
        <button
          class="chakra-button css-8hre7r"
          type="button"
        >
          liquidity.common.cancel
        </button>
        <button
          class="chakra-button css-tdyi57"
          type="button"
        >
          liquidity.common.confirm
        </button>
      </div>
    </div>
  </div>
</div>
`;
